src/core/bitop.d

```diff
@@ -396,6 +396,32 @@ struct BitRange
         }
     }
 
+    /**
+     * Construct a BitRange.
+     *
+     * Params:
+     *   bitarr - The array of bits to iterate over
+     *   numBits - The total number of valid bits in the given bit array
+     *   startBit - The initial start index to start searching
+     */
+    this(const(size_t)* bitarr, size_t numBits, size_t startBit) @system
```

Comment:
Hm.. this could be done I think with a default startBit arg instead of a new function.

Reply:
I didn't want to change the performance or inlinability of the other ctor.

Reply:
Possibly the inlinability may be affected, but it is the ctor and not the hot path. But it made me pause a bit to see so much duplicated code :)

Reply:
The ctor can also be in a hot path. But you are right, it's not in the GC mark function. Actually, using the BitRange proved to be slower than testing single bits for possible pointers on my systems, so it's disabled via `enum useBitRange = false;`. @Jebbs measured improvements with the BitRange, though.

Reply:
> Actually, using the BitRange proved to be slower than testing single bits for possible pointers on my systems

I wrote the `BitRange` code to help me avoid the tedious task of finding all set bits in a bitset when checking for module cycles. I didn't spend really any time profiling, just used tricks I knew of, though it turned out to be faster than the previous method of foreaching over the bits and checking if they were set. Would be cool if we could improve it for your purposes. It probably depends highly on how dense the pointers are.

Reply:
> It probably depends highly on how dense the pointers are.

I guess so, too. The BitRange probably helps if it allows skipping a full cache line, but that doesn't seem to happen very often in our benchmarks.